### ISSUE TITLE
Exact Same Code Yields Different Output Among Peers 

### ISSUE THREAD

**Issue by aubreymartinez** [ANTsX/ANTs#1084]
My peers and I are experiencing some strange difficulties when working with the Atropos command. We all work with ANTs (installed and compiled the program synchronously) on Laptops with a Mac operating system (10.15.6 for the two of us with the issue, and 10.15.5 for the one without). We all name our test files exactly the same so we can run the exact same code: 

antsAtroposN4.sh -d 3 -a ./bcorr_bet_Kuopio30d.nii.gz -x ./mask.nii.gz -c 3 -o output

When a colleague and I run the code above, the command only prints out a text file and a single image with 3 tissue classes. However, when our post-doc runs the exact same code, the command pints out a text file, a single image with 3 tissue classes, a grey matter segmentation image (SegmentationPosteriors2), a white matter segmentation image (SegmentationPosteriors3), and a CSF segmentation image (SegmentationPosteriors1). 

We cannot figure out why this is discrepancy is occurring, and are seeking assistance.

**Comment by ntustison** [comment]
These system-specific issues are extremely hard for us to debug and the information above is not sufficient to provide any suggestions as to the cause.  This script has been vetted for quite some time and the core developers all use Macs so I'm guessing there's something going on with your (and your colleague's) system as the number of outputs does not appear to be correct.  I would take a look at the screen print out and see if any error messages are appearing or any differences in screen output.  Next would be checking for obvious discrepancies like shell choice and possibly using full paths to specify the inputs/outputs.

**Comment by aubreymartinez** [comment]
> These system-specific issues are extremely hard for us to debug and the information above is not sufficient to provide any suggestions as to the cause. This script has been vetted for quite some time and the core developers all use Macs so I'm guessing there's something going on with your (and your colleague's) system as the number of outputs does not appear to be correct. I would take a look at the screen print out and see if any error messages are appearing or any differences in screen output. Next would be checking for obvious discrepancies like shell choice and possibly using full paths to specify the inputs/outputs.

The error messages show that the temporary files keep getting deleted.
We have tried using specifying full paths but the issues persisted.

**Comment by ntustison** [comment]
It's too bad that you weren't able to resolve the issue.  Unfortunately, as I mentioned previously, there's very little that we can do for these system-specific issues.  I use ANTs on multiple Apple machines as well as Linux and I haven't seen this issue before.  You mention that "The error messages show that the temporary files keep getting deleted."  Since temporary files are created and deleted as part of the normal processing by this script, I'm doubtful that this is an error.  I would go through *line-by-line* (a ``diff`` of some sort would probably be sufficient) for each of the screen outputs and determine what is different.

**Comment by cookpa** [comment]
I tried this script on two Macs with 10.15.6, one using the bash shell and another using zsh. I got posteriors correctly for both. As Nick says, when something works on a lot of machines but not a particular machine, it's very likely something specific to that particular machine.

A reproducible example with the command, data, the full terminal output, something like

```
antsAtroposN4.sh -d 3 -a ./bcorr_bet_Kuopio30d.nii.gz -x ./mask.nii.gz -c 3 -o output > terminalOutput.txt 2>&1
```

might provide some clues. That's why we ask for all this information in the issue template.
